.github/actions/setup-java/action.yml

```
@@ -21,7 +21,16 @@ runs:
       uses: stCarolas/setup-maven@07fbbe97d97ef44336b7382563d66743297e442f # v4.5
       with:
         maven-version: 3.9.3
-    
+
+    - name: Cache local Maven repository
+      uses: actions/cache@v3
+      with:
+        path: ~/.m2/repository
+        # re-cache on changes in the pom and target files
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml', '**/*.target') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
```

I guess this would allow for a bit more intelligent cache matching.
```suggestion
        restore-keys: |
          ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml', '**/*.target') }}
          ${{ runner.os }}-maven-
```